artichoke-backend/src/extn/core/symbol/symbol.rb

```diff
@@ -74,7 +74,18 @@ def swapcase
   end
 
   def to_proc
-    ->(obj, *args, &block) { obj.__send__(self, *args, &block) }
+    pr = lambda do |*args, &block|
+      raise ArgumentError, 'no receiver given' if args.empty?
+
+      obj, *rest = *args
+      obj.__send__(self, *rest, &block)
+    end
+
+    def pr.parameters
```

Comment:
this is gross and probably means this constructor needs to move to Rust and be represented by an enum somehow.